base/applications/cmdutils/more/more.c

```diff
@@ -106,25 +253,92 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
     dwMode &= ~ENABLE_PROCESSED_INPUT;
     SetConsoleMode(hInput, dwMode);
 
-    do
+    // FIXME: Does not support TTY yet!
+    for (;;)
     {
-        // FIXME: Does not support TTY yet!
-
-        // ConInKey(&KeyEvent);
-        INPUT_RECORD ir;
+        INPUT_RECORD ir = {0};
         DWORD dwRead;
+        WCHAR ch;
+
         do
         {
             ReadConsoleInput(hInput, &ir, 1, &dwRead);
         }
         while ((ir.EventType != KEY_EVENT) || (!ir.Event.KeyEvent.bKeyDown));
 
-        /* Got our key, return to caller */
+        /* Got our key */
         KeyEvent = ir.Event.KeyEvent;
+
+        /* Ignore any unsupported keyboard press */
+        if ((KeyEvent.wVirtualKeyCode == VK_SHIFT) ||
+            (KeyEvent.wVirtualKeyCode == VK_MENU)  ||
+            (KeyEvent.wVirtualKeyCode == VK_CONTROL))
+        {
+            continue;
+        }
```

Comment:
This has been moved up compared to what was in your code, to have a behaviour closer to the original (original in master) one.

Reply:
OK.